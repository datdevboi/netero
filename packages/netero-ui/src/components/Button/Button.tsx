import * as React from "react";
import styled from "styled-components";
import { darken } from "polished";

const btnSizes: { [key: string]: string } = {
  sm: "100px",
  md: "120px",
  lg: "250px"
};

const btnTypes: { [key: string]: string } = {
  Primary: "#007aff",
  Info: "#62B1F6",
  Success: "#5cb85c",
  Danger: "#d9534f",
  Warning: "#f0ad4e"
};

export interface IBtnProps {
  color?: string;
  btnType?: "Primary" | "Info" | "Success" | "Danger" | "Warning";
  size?: "sm" | "md" | "lg";
  className?: string;
  type?: "submit" | "button" | "reset";
  onClick: () => void;
  children: any;
}

const Btn = styled.button<IBtnProps>`
  background: ${({ color, btnType }) =>
    color ? color : btnTypes[btnType || "Primary"]};
  
 
  width: ${props => btnSizes[props.size || "md"]};
  height: auto; 
  border: none;
  border-radius: 2px;
  padding: 7px 7px;
  color: white;
  font-size: auto;
  letter-spacing: .5px;
  font-weight: normal;
  box-shadow: 5px 5px 7px #808080;  
  font-family: Helvetica, sans-serif;
  
  
  
}
  &:hover {
    background: ${({ color, btnType }) =>
      color
        ? darken(0.07)(color)
        : darken(0.07)(btnTypes[btnType || "Primary"])};
   
    letter-spacing: .5px;
    cursor: pointer;
  }
  &:active {
    box-shadow: 3px 2px 8px #808080;  
    outline: none;
  }
  
  &:disabled {
    background: grey;
    box-shadow: none;
    letter-spacing: normal;
    cursor: default;
  }

 
`;

export class Button extends React.Component<IBtnProps> {
  static defaultProps = {
    btnType: "Primary",
    size: "md"
  };
  render() {
    const { children, ...otherProps } = this.props;
    return <Btn {...otherProps}>{children}</Btn>;
  }
}
