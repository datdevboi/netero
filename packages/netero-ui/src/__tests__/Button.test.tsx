import React from "react";
import {
  render,
  fireEvent,
  cleanup,
  waitForElement
} from "react-testing-library";
import { Button } from "../components/Button/Button";

describe("Button", () => {
  const myMockOnClick = jest.fn();
  const { container, getByText } = render(
    <Button onClick={myMockOnClick}>Login</Button>
  );

  test("it renders", () => {
    expect(container).toMatchSnapshot();
  });

  test("test onClick", () => {
    fireEvent(
      getByText("Login"),
      new MouseEvent("click", {
        bubbles: true,
        cancelable: true
      })
    );

    expect(myMockOnClick).toHaveBeenCalledTimes(1);
  });
});
