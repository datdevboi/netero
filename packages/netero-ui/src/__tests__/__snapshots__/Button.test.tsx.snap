// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button it renders 1`] = `
<div>
  <button
    class="sc-bdVaJa fnUURf"
  >
    Login
  </button>
</div>
`;
